main_test.go

```diff
@@ -343,17 +363,19 @@ func Test_runVerify(t *testing.T) {
 		tt := tt // Re-initializing variable so it is not changed while executing the closure below
 		t.Run(tt.name, func(t *testing.T) {
 			// t.Parallel()
-			getBuildersAndVersions := func() []string {
+			getBuildersAndVersions := func(minversion string, ttBuilders []string) []string {
 				res := []string{}
-				builders := []string{}
-				testdataDir, err := ioutil.ReadDir("./testdata")
-				if err != nil {
-					t.Fatal(err)
-				}
-				for _, f := range testdataDir {
-					if f.IsDir() {
-						// These are the builder subfolders
-						builders = append(builders, f.Name())
+				builders := tt.builders
+				if len(builders) == 0 {
+					testdataDir, err := ioutil.ReadDir("./testdata")
```

Comment:
A suggestion would be to have `./testdata` as a constant.

The same  is being used here in next few lines below `ioutil.ReadDir(fmt.Sprintf("./testdata/%s", builder))`